pipeline/update-ips.bats

```diff
@@ -0,0 +1,1088 @@
+
+  assert_equal "$(mock_get_call_num "${mock_dig}")" 3
+  assert_equal "$(mock_get_call_num "${mock_kubectl}")" 12
+  assert_equal "$(mock_get_call_num "${mock_curl}")" 0
+}
+
+#
+# SWIFT
+#
+# if [ "$check_harbor" == "swift" ] || [ "$check_thanos" == "swift" ] || [ "${sourceSwift}" == "true" ];
+#
+# $check_harbor: '.harbor.persistence.type'
+# $check_thanos: '.thanos.objectStorage.type'
+# $sourceSwift:  '.objectStorage.sync.buckets.[].sourceType' == 'swift'
+#
+
+_test_swift_requires_openstack_auth_endpoint() {
+  _setup_base
+
+  yq4 -i "${1}"' = "swift"' "${CK8S_CONFIG_PATH}/sc-config.yaml"
+
+  run ck8s update-ips both dry-run
+  assert_failure
+  assert_output --partial "No openstack auth endpoint found"
+
+  assert_equal "$(mock_get_call_num "${mock_dig}")" 3
+  assert_equal "$(mock_get_call_num "${mock_kubectl}")" 12
+  assert_equal "$(mock_get_call_num "${mock_curl}")" 0
+}
+
+@test "ck8s update-ips both dry-run - swift requires openstack auth endpoint - harbor" {
+  _test_swift_requires_openstack_auth_endpoint '.harbor.persistence.type'
+}
+
+@test "ck8s update-ips both dry-run - swift requires openstack auth endpoint - thanos" {
+  _test_swift_requires_openstack_auth_endpoint '.thanos.objectStorage.type'
+}
+
+@test "ck8s update-ips both dry-run - swift requires openstack auth endpoint - bucket source type" {
+  _test_swift_requires_openstack_auth_endpoint '.objectStorage.sync.buckets.[0].sourceType'
+}
+
+_test_swift_requires_swift_region() {
+  _setup_base
+
+  yq4 -i "${1}"' = "swift"' "${CK8S_CONFIG_PATH}/sc-config.yaml"
+  yq4 -i '.objectStorage.swift.authUrl = "https://keystone.foo.dev-ck8s.com"' "${CK8S_CONFIG_PATH}/sc-config.yaml"
+
+  mock_set_output "${mock_dig}" "127.0.0.4" 4 # $os_auth_endpoint
+
+  run ck8s update-ips both dry-run
+  assert_failure
+  assert_output --partial "No Swift region found"
+
+  assert_equal "$(mock_get_call_num "${mock_dig}")" 4
+  assert_equal "$(mock_get_call_num "${mock_kubectl}")" 12
+  assert_equal "$(mock_get_call_num "${mock_curl}")" 0
+}
+
+@test "ck8s update-ips both dry-run - swift requires swift region - harbor" {
+  _test_swift_requires_swift_region '.harbor.persistence.type'
+}
+
+@test "ck8s update-ips both dry-run - swift requires swift region - thanos" {
+  _test_swift_requires_swift_region '.thanos.objectStorage.type'
+}
+
+@test "ck8s update-ips both dry-run - swift requires swift region - bucket source type" {
+  _test_swift_requires_swift_region '.objectStorage.sync.buckets.[0].sourceType'
+}
+
+_test_swift_requires_username_or_application_credential_id() {
+  _setup_base
+
+  yq4 -i "${1}"' = "swift"' "${CK8S_CONFIG_PATH}/sc-config.yaml"
+  yq4 -i '.objectStorage.swift.authUrl = "https://keystone.foo.dev-ck8s.com"' "${CK8S_CONFIG_PATH}/sc-config.yaml"
+  yq4 -i '.objectStorage.swift.region = "swift-region"' "${CK8S_CONFIG_PATH}/sc-config.yaml"
+
+  mock_set_output "${mock_dig}" "127.0.0.4" 4 # $os_auth_endpoint
+
+  run ck8s update-ips both dry-run
```

Comment:
Any reason for using `update-ips both` when you seemingly only use sc-config?

Reply:
I decided to only add tests for `update-ips both` as it should hopefully add test coverage for all cases. Do you know if there is any parts of the script that is not evaluated if you run `both`?

Reply:
Everything should be evaluated when you use `both` i was just thinking that it would take longer to go through it all when your using `both` instead of `sc` seeing as most of the script you are only evaluating `sc`.

Reply:
Ah I see! It's possible that the test can be optimized once the script is refactored. Since it was completely new to me I just went the safest route possible to begin with. I'll keep it in mind, thanks!